File: examples/test_class_methods.rs
use serde_json;
#[derive(Debug, Clone)]
pub struct Calculator {
    pub value: i32,
    pub history: Vec<serde_json::Value>,
}
impl Calculator {
    pub fn new(initial_value: i32) -> Self {
        Self {
            value: 0,
            history: Vec::new(),
        }
    }
    pub fn add(&mut self, x: i32) -> i32 {
        self.value = self.value + x;
        self.history.push(format!("add({})", x));
        return self.value;
    }
    pub fn multiply(&mut self, x: i32) -> i32 {
        self.value = self.value * x;
        self.history.push(format!("multiply({})", x));
        return self.value;
    }
    pub fn square(x: i32) -> i32 {
        return x * x;
    }
    pub fn from_string(s: String) {
        return Self::new(s.parse::<i32>().unwrap_or(0));
    }
    pub fn current(&self) -> i32 {
        return self.value;
    }
    pub fn get_history(&self) -> Vec<String> {
        return self.history.clone();
    }
}
#[doc = " Depyler: verified panic-free"]
#[doc = " Depyler: proven to terminate"]
pub fn test_calculator() -> (
    serde_json::Value,
    serde_json::Value,
    serde_json::Value,
    serde_json::Value,
    serde_json::Value,
) {
    let mut calc = Calculator::new(10);
    let result1 = calc.add(5);
    let result2 = calc.multiply(2);
    let squared = Calculator::square(4);
    let current = calc.current;
    let history = calc.get_history();
    (result1, result2, squared, current, history)
}

File: examples/test_class_methods.py
"""Test class with various method types for v1.2.0"""

class Calculator:
    """A simple calculator class demonstrating method types"""
    
    def __init__(self, initial_value: int = 0):
        """Initialize calculator with a value"""
        self.value = initial_value
        self.history: list[str] = []
    
    def add(self, x: int) -> int:
        """Add to current value (instance method)"""
        self.value += x
        self.history.append(f"add({x})")
        return self.value
    
    def multiply(self, x: int) -> int:
        """Multiply current value (instance method)"""
        self.value *= x
        self.history.append(f"multiply({x})")
        return self.value
    
    @staticmethod
    def square(x: int) -> int:
        """Square a number (static method)"""
        return x * x
    
    @classmethod
    def from_string(cls, s: str):
        """Create calculator from string (class method)"""
        return cls(int(s))
    
    @property
    def current(self) -> int:
        """Get current value (property)"""
        return self.value
    
    def get_history(self) -> list[str]:
        """Get operation history"""
        return self.history.copy()

# Test usage
def test_calculator():
    # Create instance
    calc = Calculator(10)
    
    # Instance methods
    result1 = calc.add(5)
    result2 = calc.multiply(2)
    
    # Static method
    squared = Calculator.square(4)
    
    # Property access
    current = calc.current
    
    # Get history
    history = calc.get_history()
    
    return result1, result2, squared, current, history